server/src/test/java/org/elasticsearch/common/settings/ScopedSettingsTests.java

```diff
@@ -1221,4 +1256,54 @@ public String getKey(final String key) {
                 equalTo(oldSetting.get(settings).stream().map(s -> "new." + s).collect(Collectors.toList())));
     }
 
+    public void testUpdateOfValidationDependantSettings() {
+        ClusterSettings service = new ClusterSettings(Settings.EMPTY, new HashSet<>(Arrays.asList(SETTING_FOO_LOW, SETTING_FOO_HIGH)));
+
+        Settings.Builder target = Settings.builder();
+        Settings.Builder updates;
+        boolean updated;
+
+        updates = Settings.builder();
+        updated = service.updateSettings(
+            Settings.builder().put(SETTING_FOO_LOW.getKey(), 20).build(),
+            target,
+            updates,
+            "transient"
+        );
+        assertTrue(updated);
+        assertThat(target.get(SETTING_FOO_LOW.getKey()), equalTo("20"));
+        assertThat(updates.get(SETTING_FOO_LOW.getKey()), equalTo("20"));
```

Comment:
We can write this with fewer lines and fewer mutable variables by putting each of these steps into its own block like this:

```
        {
            final Settings.Builder updates = Settings.builder();
            assertTrue(service.updateSettings(Settings.builder().put(SETTING_FOO_LOW.getKey(), 20).build(), target, updates, "transient"));
            assertThat(target.get(SETTING_FOO_LOW.getKey()), equalTo("20"));
            assertThat(updates.get(SETTING_FOO_LOW.getKey()), equalTo("20"));
        }
```

The line-width limit is 140 characters, and this fits into that very neatly. It's something of a matter of taste, but we seem to prefer denser code in this sort of situation.

Reply:
That's fine for me. The more immutable code is, the more confident I feel.